### PR TITLE
Add documentation and implementation reference for internal AA hash.

### PR DESCRIPTION
**Pull request by ibuclaw**
Information gathered from comments in #2436 after algorithm was introduced.

### CODE REVIEW

**Review by JinShil — APPROVED** [review]

File: src/dmd/root/aav.d [hunk 1]
```diff
@@ -15,29 +15,46 @@ module dmd.root.aav;
...
 struct AA
 {
     aaA** b;
     size_t b_length;
-    size_t nodes; // total number of aaA nodes
-    aaA*[4] binit; // initial value of b[]
-    aaA aafirst; // a lot of these AA's have only one entry
+    size_t nodes;   // total number of aaA nodes
+    aaA*[4] binit;  // initial value of b[]
+    aaA aafirst;    // a lot of these AA's have only one entry
```
Comment (jacob-carlborg) [inline src/dmd/root/aav.d]: I prefer to not align. The problem is if/when a new line is added it's necessary to realign all lines. Which is more work and creates more noisy diffs.

Comment (ibuclaw) [reply src/dmd/root/aav.d]: Doesn't make for easy reading though having it unaligned.

**Comment by ibuclaw** [comment]
Although on the flip side, I just came across #5428.  So maybe should be doing the opposite of documenting this.

**Comment by ibuclaw** [comment]
Yeah, I changed my mind. #7694

**Pull request closed without merging**